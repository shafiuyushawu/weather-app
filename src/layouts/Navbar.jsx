import { Link } from 'react-router-dom';
import { FcHome } from 'react-icons/fc';

const Navbar = () => (
  <nav className="navbar bg-neutral text-neutral-content" data-test="navbar">
    <div className="flex-1">
      <Link to="/" className="btn btn-ghost normal-case text-xl p-2">
        <FcHome className="inline text-4xl mr-4" />
        Weather Forecast
      </Link>
    </div>
  </nav>
);

export default Navbar;
