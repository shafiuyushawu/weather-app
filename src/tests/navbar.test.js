import { MemoryRouter, BrowserRouter as Router } from 'react-router-dom';
import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import Navbar from '../layouts/Navbar';
import '@testing-library/jest-dom/extend-expect';

describe('Navbar', () => {
  it('matches snapshot', () => {
    const tree = renderer
      .create(
        <Router>
          <Navbar />
        </Router>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders Weather Forecast in the Navbar', () => {
    const { getByText } = render(
      <MemoryRouter>
        <Navbar />
      </MemoryRouter>,
    );

    const linkElement = getByText(/Weather Forecast/i);
    expect(linkElement).toBeInTheDocument();
  });
});
