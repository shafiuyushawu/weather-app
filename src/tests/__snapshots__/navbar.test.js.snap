// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar matches snapshot 1`] = `
<nav
  className="navbar bg-neutral text-neutral-content"
  data-test="navbar"
>
  <div
    className="flex-1"
  >
    <a
      className="btn btn-ghost normal-case text-xl p-2"
      href="/"
      onClick={[Function]}
    >
      <svg
        className="inline text-4xl mr-4"
        enableBackground="new 0 0 48 48"
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        version="1"
        viewBox="0 0 48 48"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <polygon
          fill="#E8EAF6"
          points="42,39 6,39 6,23 24,6 42,23"
        />
        <g
          fill="#C5CAE9"
        >
          <polygon
            points="39,21 34,16 34,9 39,9"
          />
          <rect
            height="5"
            width="36"
            x="6"
            y="39"
          />
        </g>
        <polygon
          fill="#B71C1C"
          points="24,4.3 4,22.9 6,25.1 24,8.4 42,25.1 44,22.9"
        />
        <rect
          fill="#D84315"
          height="16"
          width="12"
          x="18"
          y="28"
        />
        <rect
          fill="#01579B"
          height="6"
          width="6"
          x="21"
          y="17"
        />
        <path
          d="M27.5,35.5c-0.3,0-0.5,0.2-0.5,0.5v2c0,0.3,0.2,0.5,0.5,0.5S28,38.3,28,38v-2C28,35.7,27.8,35.5,27.5,35.5z"
          fill="#FF8A65"
        />
      </svg>
      Weather Forecast
    </a>
  </div>
</nav>
`;
